release.js

```diff
@@ -55,12 +57,31 @@ module.exports = async function ({ github, context, inputs, callApi }) {
 
   // TODO: What if PR was closed, reopened and then merged. The draft release would have been deleted!
 
-  await callApi({
-    endpoint: 'release',
-    method: 'PATCH',
-    body: {
-      version: version,
-      releaseId: id,
-    },
-  })
+  try {
+    await callApi({
+      endpoint: 'release',
+      method: 'PATCH',
+      body: {
+        version: version,
+        releaseId: id,
+      },
+    })
+
+    try {
+      const syncVersions = /true/i.test(inputs['sync-semver-tags'])
+
+      if (syncVersions) {
+        const parsed = semver.parse(version)
+        const major = parsed.major
+        const minor = parsed.minor
+
+        if (major !== 0) await tagVersionInGit(`v${major}`)
+        if (minor !== 0) await tagVersionInGit(`v${major}.${minor}`)
+      }
+    } catch (err) {
+      core.setFailed(`Unable to update the semver tags ${err.message}`)
+    }
```

Comment:
Wondering if this should happen regardless of if publishing the release failed. If the maintainer sets the draft release as published manually after this action runs, the major and minors tags would already be updated.

Reply:
no right or wrong here, but I do agree that it may make sense to do the tagging anyway. then let's move the tagging before the release creation.

let's also avoid nested try/catch please